src/renderer/components/RecentProjectsBlock.tsx

```diff
@@ -1,36 +1,121 @@
-import { ApplicationStore } from '../store/helpers';
-import GenericSquareBox from './GenericSquareBox';
+import { Box, Typography, styled, Stack, Grid, Paper } from '@mui/material';
+import { useDispatch, useSelector } from 'react-redux';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { projectOpened, pageChanged } from 'renderer/store/actions';
+import { ApplicationPage, ApplicationStore } from '../store/helpers';
+import colors from '../colors';
+import { formatDate } from '../util';
+import exampleThumbnail from '../../../assets/example-thumbnail.png';
 
 const RecentProjectsBox = styled(Box)`
-  background: ${colors.grey[900]};
-  width: calc(100vw - 80px);
+  width: calc(100vw - 40px);
   margin: 20px;
   padding: 20px;
 `;
 
-const ProjectsListBox = styled(Box)`
-  margin-top: 20px;
+const RecentProjectsItem = styled(Paper)`
+  background: ${colors.grey[700]};
+  color: ${colors.grey[300]};
+  padding: 10px 20px;
+  padding-right: 0;
+
+  transition: 0.5s background;
+
+  &:hover {
+    background: ${colors.grey[600]};
+    cursor: pointer;
+  }
+`;
+
+const RecentProjectsSubItem = styled(Grid)`
   display: flex;
-  overflow-x: auto;
+  align-items: center;
+  justify-items: right;
 `;
 
+// No idea why, but styling typography in paper in the usual way just literally does nothing. Seems like an MUI bug,
+// can't find any documentation on it. So doing this instead
+const CategoryHeading = ({ children }: { children: React.ReactNode }) => (
+  <Typography style={{ fontSize: 12, color: colors.grey[400] }}>
+    {children}
+  </Typography>
+);
+
+const sortByDateModified = (first: RecentProject, second: RecentProject) =>
+  (second.dateModified?.getTime() || 0) - (first.dateModified?.getTime() || 0);
+
+const RECENT_PROJECTS_COUNT = 5;
+
 const RecentProjectsBlock = () => {
+  const dispatch = useDispatch();
+
   const recentProjects = useSelector(
     (store: ApplicationStore) => store.recentProjects
-  );
+  )
+    .sort(sortByDateModified)
+    .slice(0, RECENT_PROJECTS_COUNT);
+
+  const displayDate: (date: Date | null) => string = (date) =>
+    date === null ? '?' : formatDate(date);
+
+  const formatSize: (size: number | null) => string = (size) =>
+    size === null ? '?' : `${Math.floor(size / 1000000)} MB`;
+
+  const openProject: (id: string) => void = (id) => {
+    const project = recentProjects.find((proj) => proj.id === id);
+
+    if (!project) {
+      return;
+    }
+
+    dispatch(projectOpened(project, project.projectFilePath));
+    dispatch(pageChanged(ApplicationPage.PROJECT));
+  };
 
   return (
     <RecentProjectsBox>
-      <Typography fontWeight="bold">Recent Projects</Typography>
-      <ProjectsListBox>
-        {recentProjects.map(({ id, name }) => (
-          <GenericSquareBox key={id}>
-            <Typography fontWeight="bold">{name}</Typography>
-          </GenericSquareBox>
+      <Typography variant="h6" style={{ marginBottom: 20, fontWeight: 'bold' }}>
```

Comment:
Use font styles defined by @hhen0005 in theme.tsx - I believe this is h3

Reply:
H3 is too small, let's do what appears right for now and match it up once the sizes are properly defined

Reply:
Figma has defined everything - do the sizes/measurements given on Figma look different when you try to implement them?

Reply:
Okay, initially I thought this was just because the figma frame size was quite small (height = 752, 13 inch macbook = 900 and most monitors higher)

But, actually it was because the theme variants were not working and everything was defaulting to 16px when I used them.

I've fixed them though this has involved ditching the whole 'h-100' and 'h-300' idea - MUI creates elements with those types which is not a good idea since they are not actually html element types! I've changed it back to use h1 and h3. @hhen0005 please review, you're more across this theming stuff than I am

Reply:
Anyway, using h1 and h3 now and they look fine

Reply:
Annoying that it doesn't work as advertised. I understand the reason for changing. Can you also please remove the type declarations for h-300 and h-100 if they no longer exist (if you haven't done so already) :). Thanks